editor/actions.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { parse, stringify } from 'querystring';
+
+export function savePost( dispatch, postId, edits ) {
+	const toSend = postId ? { id: postId, ...edits } : edits;
+	const isNew = ! postId;
+
+	dispatch( {
+		type: 'REQUEST_POST_UPDATE',
+		edits,
+		isNew,
+	} );
+
+	new wp.api.models.Post( toSend ).save().done( ( newPost ) => {
+		dispatch( {
+			type: 'REQUEST_POST_UPDATE_SUCCESS',
+			post: newPost,
+			isNew,
+		} );
+		if ( isNew && window.history.replaceState ) {
+			const urlPieces = window.location.href.split( '?' );
+			const qs = parse( urlPieces[ 1 ] || '' );
+			const newUrl = urlPieces[ 0 ] + '?' + stringify( {
+				...qs,
```

Comment:
I tried this, and I think the code clarity / elegance due to using the spread operator is worth the extra clone step.  I don't think the performance is a concern here as this is not called very often and the browser's `replaceState` is going to be orders of magnitude more expensive than the `assign`.